pydbg.py

```diff
@@ -24,11 +31,17 @@ def square(x: int) -> int:
 
     """
 
-    for i in reversed(inspect.stack()):
-        ctx = i.code_context[0]
-        if "dbg" in ctx:
+    for frame in inspect.stack():
```

Comment:
Any reason why the "reversed" was removed? Usually we want stuff at the end of the stack rather than the beginning so we can get an early exit

Reply:
From https://docs.python.org/3/library/inspect.html#inspect.stack:

> The first entry in the returned list represents the caller; the last entry represents the outermost call on the stack.

So, I believe we want to start from `[0]`. Also, as explained in one of the issues, I believe that's all we need and shouldn't go up the stack at all.

Reply:
I think we really don't want to iterate over it all. I think we want the `[1]` element from the stack. The `[0]` elements will always be the literal `dbg` function in `pydbg` and the second elements on the stack will be the call that we want.

Reply:
Probably could be better if it was like: 

```
line = inspect.stack()[1].code_context[0]
start = line.find('(') + 1
end =  line.rfind(')')
if end == -1:
    end = len(line)
print(
    f"[{frame.filename}:{frame.lineno}] {line[start:end]} = {exp!r}",
    file=sys.stderr,
)
```

Reply:
Yeah, I think that's correct. We want `[1]`, not `[0]`. Anyways, I left it for another patch, which would hopefully also support handling multi-line `dbg(...)` calls. Can get to that next week or so, if no one else is working on it.

Reply:
Sounds good to me. I might have some time to take a stab on it next week as I am traveling. 

Thanks @besfahbod